pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -0,0 +1,706 @@
+// Copyright (c) 2019 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Flow Statistics collection from IPtable Conntrack facility
+
+package zedrouter
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/eriknordmark/netlink"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"github.com/lf-edge/eve/pkg/pillar/cast"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/wrap"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"syscall"
+)
+
+type flowStats struct {
+	SrcIP       net.IP
+	DstIP       net.IP
+	SrcPort     uint16
+	DstPort     uint16
+	Proto       uint8
+	SendPkts    uint64
+	SendBytes   uint64
+	RecvPkts    uint64
+	RecvBytes   uint64
+	TimeStart   uint64
+	TimeStop    uint64
+	TimeOut     uint32
+	aclNum      uint32
+	appNum      uint8
+	AppInitiate bool
+	IsTimeOut   bool
+	dbg1        int
+	dbg2        int
+}
+
+type aclAttr struct {
+	aclNum    uint32
+	tableName string
+	aclName   string
+	chainName string
+	action    string
+	vif       string
+	bridge    string
+	netUUID   uuid.UUID
+}
+
+type networkAttrs struct {
+	ipaclattr  map[int]map[int]aclAttr // appNum, ACLNum, acl attributes
+	appIPAddrs map[int][]net.IP        // appNum, IP addresses (may belong to diff bridges)
+	intfAddrs  []net.Addr              // device interface addresses
+	bnNet      map[string]aclAttr      // mainly need to range all the bridge interfaces
+	appNet     map[int]uuid.UUID       // max 256 apps
+}
+
+type dnsEntry struct {
+	AppIP      net.IP    // DNS message replies to
+	DomainName string    // Domain Name
+	TimeStamp  time.Time // time of the DNS server reply
+	isIPv4     bool      // returned IP is protocol IPv4
+	ANCount    uint16    // number of IP addresses in reply
+	Answers    []net.IP  // list of IP addresses
+}
+
+const (
+	maxBridgeNumber int    = 256
+	timeoutSec      int32  = 150    // less than 150 sec, consider done
```

Comment:
Might make sense to make this variable configurable from configuration. It can help easily tweak timeout value for different network traffic situations. Not sure if this should be done now or create a story and work on it later.

Reply:
we can create a story on this.